internal/rule/escalation.go

```diff
@@ -21,6 +22,30 @@ type Escalation struct {
 	Recipients []*EscalationRecipient
 }
 
+// MarshalLogObject implements the zapcore.ObjectMarshaler interface.
+//
+// This allows us to use `zap.Inline(escalation)` or `zap.Object("rule_escalation", escalation)` wherever
+// fine-grained logging context is needed, without having to add all the individual fields ourselves each time.
+// https://pkg.go.dev/go.uber.org/zap/zapcore#ObjectMarshaler
+func (e *Escalation) MarshalLogObject(encoder zapcore.ObjectEncoder) error {
+	encoder.AddInt64("id", e.ID)
+	encoder.AddInt64("rule_id", e.RuleID)
+
+	if e.NameRaw.Valid && e.NameRaw.String != "" {
+		encoder.AddString("name", e.NameRaw.String)
+	} else {
+		encoder.AddString("name", e.DisplayName())
+	}
```

Comment:
Why does this need to access `NameRaw` here instead of just relying on `DisplayName()`?

Reply:
Because if `zap.Inline(escalation)` is used from within the runtime config, `escalation.Name` could still be empty, while `NameRaw` is not. 
https://github.com/Icinga/icinga-notifications/blob/454502ef05c86e944f14705f896710ab03e1fc44/internal/config/rule.go#L90-L92

Reply:
The `Name` attribute is only used in `DisplayName()` and where it's initialized, so it really isn't needed (for where the name is needed, `DisplayName()` should be used). Thus, I'd just remove the current `Name` member, and move what's done here into `DisplayName()`.

Reply:
Done!